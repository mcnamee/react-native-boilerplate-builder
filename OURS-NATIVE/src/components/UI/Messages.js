import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { Text } from 'native-base';

import Colors from '../../../native-base-theme/variables/commonColor';

const Messages = ({ message, type }) => (
  <View
    style={{
      backgroundColor:
        type === 'error' ? Colors.brandDanger : type === 'success' ? Colors.brandSuccess : Colors.brandInfo,
      paddingVertical: 10,
      paddingHorizontal: 5,
    }}
  >
    <Text style={{ color: '#fff', textAlign: 'center' }}>{message}</Text>
  </View>
);

Messages.propTypes = {
  message: PropTypes.string,
  type: PropTypes.oneOf(['error', 'success', 'info']),
};

Messages.defaultProps = {
  message: 'An unexpected error came up',
  type: 'error',
};

export default Messages;
