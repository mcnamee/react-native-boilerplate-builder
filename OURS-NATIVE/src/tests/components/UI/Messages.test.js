import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react-native';
import Messages from '../../../components/UI/Messages';

it('<Messages /> renders with error and message', () => {
  const Component = <Messages message="Success" type="success" />;

  // Matches snapshot
  expect(renderer.create(Component).toJSON()).toMatchSnapshot();

  // Has the correct text on the page
  const { getByText } = render(Component);
  expect(getByText('Success'));
});

it('<Messages /> renders with error and message', () => {
  const Component = <Messages message="Error" type="error" />;

  // Matches snapshot
  expect(renderer.create(Component).toJSON()).toMatchSnapshot();

  // Has the correct text on the page
  const { getByText } = render(Component);
  expect(getByText('Error'));
});

it('<Messages /> renders with info and message', () => {
  const Component = <Messages message="Warning" type="info" />;

  // Matches snapshot
  expect(renderer.create(Component).toJSON()).toMatchSnapshot();

  // Has the correct text on the page
  const { getByText } = render(Component);
  expect(getByText('Warning'));
});
