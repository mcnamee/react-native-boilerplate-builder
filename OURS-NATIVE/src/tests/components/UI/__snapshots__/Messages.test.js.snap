// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Messages /> renders with error and message 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#5cb85c",
      "paddingHorizontal": 5,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "color": "#fff",
          "textAlign": "center",
        },
      ]
    }
  >
    Success
  </Text>
</View>
`;

exports[`<Messages /> renders with error and message 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#d9534f",
      "paddingHorizontal": 5,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "color": "#fff",
          "textAlign": "center",
        },
      ]
    }
  >
    Error
  </Text>
</View>
`;

exports[`<Messages /> renders with info and message 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#62B1F6",
      "paddingHorizontal": 5,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "color": "#fff",
          "textAlign": "center",
        },
      ]
    }
  >
    Warning
  </Text>
</View>
`;
